js-wether/wether.js

```diff
@@ -0,0 +1,52 @@
+document.addEventListener("DOMContentLoaded", function () {
+    //API取得~HTML編集までの一連の動き
+    const wholeOparation = function () {
+        const xhr = new XMLHttpRequest();
+        const url = getWeatherApi();
+        xhr.open("GET", url, true);
+        xhr.onload = function () {
+            const dataJson = this.response;
+            const data = JSON.parse(dataJson);
+            const weatherEn = data.weather[0].main;
+            const weatherText = sharpData(weatherEn);
+            editHtml(weatherText);
+        };
+        xhr.send();
+    };
+
+    //APIへの問い合わせ
+    const getWeatherApi = function () {
+        //指定されたプルダウンの値を取得
+        const cityName = document.querySelector("#city").value;
+        //APIへの問い合わせURLを生成
+        const appId = "4b5774e9f3d2a07b84f0f2f88e486224";
+        const queryUrl = `http://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${appId}`;
+        return queryUrl;
+    };
+
+    //日本語の天候・町名を取得（公式ドキュメントの全天候網羅済）
+    const sharpData = function (weatherEn) {
+        //日本語の天候を取得
+        const weatherTrans = {
+            "Clouds": "曇り", "Clear": "晴天", "Snow": "雪", "Rain": "雨", "Tornado": "竜巻", "Squall": "スコール", "Thunderstorm": "雷雨", "Drizzle": "霧雨", "Mist": "霧", "Smoke": "煙っぽい", "Haze": "煙霧", "Fog": "濃霧", "Ash": "降灰", "Dust": "砂塵", "Sand": "砂っぽい", "": "不明"
+        };
```

Comment:
若干の好みもあるのですが、この状態だと本当にペアが正しいのかわかりにくいです。
こういう場合は、要素ごとに改行してしまうのが良いです。

```suggestion
        //日本語の天候を取得
        const weatherTrans = {
            "Clouds": "曇り",
            "Clear": "晴天",
            ...
```